lib/ansible/modules/cloud/amazon/rds.py

```diff
@@ -973,9 +973,10 @@ def create_db_instance(module, conn):
         changed = False
     else:
         try:
-            result = conn.create_db_instance(instance_name, module.params.get('size'),
-                                             module.params.get('instance_type'), module.params.get('db_engine'),
-                                             module.params.get('username'), module.params.get('password'), **params)
+            if not module.check_mode:
```

Comment:
Trying to create an instance with check mode fails with:
```
The full traceback is:
Traceback (most recent call last):
  File "/var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/ansible_6vt6E9/ansible_module_rds.py", line 1406, in <module>
    main()
  File "/var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/ansible_6vt6E9/ansible_module_rds.py", line 1399, in main
    invocations[module.params.get('command')](module, conn)
  File "/var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/ansible_6vt6E9/ansible_module_rds.py", line 985, in create_db_instance
    resource = await_resource(conn, result, 'available', module)
  File "/var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/ansible_6vt6E9/ansible_module_rds.py", line 929, in await_resource
    while wait_timeout > time.time() and resource.status != status:
AttributeError: 'NoneType' object has no attribute 'status'

fatal: [localhost]: FAILED! => {
    "changed": false,
    "failed": true,
    "module_stderr": "Traceback (most recent call last):\n  File \"/var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/ansible_6vt6E9/ansible_module_rds.py\", line 1406, in <module>\n    main()\n  File \"/var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/ansible_6vt6E9/ansible_module_rds.py\", line 1399, in main\n    invocations[module.params.get('command')](module, conn)\n  File \"/var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/ansible_6vt6E9/ansible_module_rds.py\", line 985, in create_db_instance\n    resource = await_resource(conn, result, 'available', module)\n  File \"/var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/ansible_6vt6E9/ansible_module_rds.py\", line 929, in await_resource\n    while wait_timeout > time.time() and resource.status != status:\nAttributeError: 'NoneType' object has no attribute 'status'\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE",
    "rc": 0
}

PLAY RECAP *******************************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1
```
because await_resource is still being called below regardless of check mode.